File: ios/LaneShadow/Sandbox/Stories/Molecules/LSContentCardStory.swift
import LaneShadowTheme
import NativeSandbox
import SwiftUI

@MainActor
enum LSContentCardStory {
    static let all: [Story] = [
        Story(
            id: "molecules.contentCard.withImageHeader",
            tier: .molecule,
            component: "LSContentCard",
            name: "With Image Header",
            summary: "Card with an optional media header above title and metadata."
        ) { _ in
            LSContentCard(
                title: "Wasatch Crest Trail",
                subtitle: "42 mi · 1h 12m",
                metadata: ["3,400 ft gain", "Last ridden Apr 18"],
                chips: ["Best Route", "Alpine"],
                header: AnyView(imageHeader),
                actions: AnyView(
                    HStack(spacing: Theme.shared.space.sm) {
                        LSButton("Ride This", variant: .primary, action: {})
                        LSButton("Save", variant: .ghost, action: {})
                    }
                )
            )
            .padding(Theme.shared.space.lg)
        },
        Story(
            id: "molecules.contentCard.titleOnly",
            tier: .molecule,
            component: "LSContentCard",
            name: "Title Only",
            summary: "Minimal title-only card surface."
        ) { _ in
            LSContentCard(title: "Emigration Canyon Climb")
                .padding(Theme.shared.space.lg)
        },
        Story(
            id: "molecules.contentCard.titleSubtitleChips",
            tier: .molecule,
            component: "LSContentCard",
            name: "Title+Subtitle+Chips",
            summary: "Content card body with subtitle, metadata, and chips."
        ) { _ in
            LSContentCard(
                title: "Emigration Canyon Climb",
                subtitle: "28 mi · 1h 04m · Mountain",
                metadata: ["3,400 ft gain", "Last ridden Apr 18"],
                chips: ["Best Route", "Alpine"]
            )
            .padding(Theme.shared.space.lg)
        },
        Story(
            id: "molecules.contentCard.withActions",
            tier: .molecule,
            component: "LSContentCard",
            name: "With Actions",
            summary: "Card with footer actions rendered beneath metadata."
        ) { _ in
            LSContentCard(
                title: "Route X",
                subtitle: "42 mi · 1h 12m",
                metadata: ["3,400 ft gain"]
            ) {
                HStack(spacing: Theme.shared.space.sm) {
                    LSButton("Ride This", variant: .primary, action: {})
                    LSButton("Save", variant: .outline, action: {})
                }
            }
            .padding(Theme.shared.space.lg)
        },
    ]

    private static var imageHeader: some View {
        ZStack(alignment: .bottomLeading) {
            RoundedRectangle(cornerRadius: Theme.shared.radius.md, style: .continuous)
                .fill(LaneShadowTheme.color.surface.glass)
                .frame(height: Theme.shared.space.xxxxl + Theme.shared.space.md)

            HStack(spacing: Theme.shared.space.xs) {
                LSIcon(name: .route, size: .sm, color: .secondary)
                LSText("Scenic preview", variant: .label.sm, color: .secondary)
            }
            .padding(.horizontal, Theme.shared.space.sm)
            .padding(.vertical, Theme.shared.space.xs)
        }
    }
}

File: android/app/src/debug/java/com/laneshadow/sandbox/stories/molecules/LSContentCardStory.kt
package com.laneshadow.sandbox.stories.molecules

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import com.laneshadow.theme.LocalLaneShadowTheme
import com.laneshadow.ui.molecules.LSContentCard
import com.nativesandbox.model.ComponentTier
import com.nativesandbox.model.Story

object LSContentCardStory {
    val all: List<Story> = listOf(
        story(
            id = "molecules.contentcard.with-image-header",
            name = "With Image Header",
            summary = "Card with optional header slot and primary text body.",
        ) {
            val theme = LocalLaneShadowTheme.current
            LSContentCard(
                title = "Wasatch Crest Trail",
                subtitle = "28 mi · 1h 04m · Mountain",
                header = {
                    Row(
                        modifier = Modifier
                            .fillMaxWidth()
                            .height(theme.sizing.icon.xl + theme.space.xxl)
                            .background(theme.colors.surfaceVariant.default)
                            .padding(theme.space.md),
                        horizontalArrangement = Arrangement.SpaceBetween,
                    ) {}
                },
            )
        },
        story(
            id = "molecules.contentcard.title-only",
            name = "Title Only",
            summary = "Minimal card body with title only.",
        ) {
            LSContentCard(
                title = "Mill Creek Canyon Loop",
            )
        },
        story(
            id = "molecules.contentcard.title-subtitle-chips",
            name = "Title+Subtitle+Chips",
            summary = "Title and subtitle with metadata chips rendered in actions slot.",
        ) {
            val theme = LocalLaneShadowTheme.current
            LSContentCard(
                title = "Emigration Canyon Climb",
                subtitle = "42 mi · 1h 12m",
                actions = {
                    Row(horizontalArrangement = Arrangement.spacedBy(theme.space.sm)) {
                        com.laneshadow.ui.atoms.LSText(
                            text = "Best Route",
                            variant = com.laneshadow.ui.atoms.TypographyVariant.Ui.Label.Sm,
                            color = com.laneshadow.ui.atoms.ContentColor.Secondary,
                        )
                        com.laneshadow.ui.atoms.LSText(
                            text = "Alpine",
                            variant = com.laneshadow.ui.atoms.TypographyVariant.Ui.Label.Sm,
                            color = com.laneshadow.ui.atoms.ContentColor.Secondary,
                        )
                    }
                },
            )
        },
        story(
            id = "molecules.contentcard.with-actions",
            name = "With Actions",
            summary = "Card footer action slot composition.",
        ) {
            val theme = LocalLaneShadowTheme.current
            LSContentCard(
                title = "Pacific Coast Sweep",
                subtitle = "82 mi · 2h 15m",
                actions = {
                    Row(horizontalArrangement = Arrangement.spacedBy(theme.space.sm)) {
                        com.laneshadow.ui.atoms.LSText(
                            text = "Ride This",
                            variant = com.laneshadow.ui.atoms.TypographyVariant.Ui.Label.Sm,
                            color = com.laneshadow.ui.atoms.ContentColor.Primary,
                        )
                        com.laneshadow.ui.atoms.LSText(
                            text = "Save",
                            variant = com.laneshadow.ui.atoms.TypographyVariant.Ui.Label.Sm,
                            color = com.laneshadow.ui.atoms.ContentColor.Secondary,
                        )
                    }
                },
            )
        },
    )
}

private fun story(
    id: String,
    name: String,
    summary: String,
    content: @Composable () -> Unit,
): Story =
    Story(
        id = id,
        tier = ComponentTier.Molecule,
        component = "LSContentCard",
        name = name,
        summary = summary,
        content = { MoleculeStoryFrame(content) },
    )
